### PR TITLE
support open file and link on WSL

### LINKED ISSUE #1996: open file and open link returns error in WSL
**Describe the bug**
opening file and opening PR returns error since xdg-open does not exist in WSL

**To Reproduce**
1. Select any branch
2. Click 'o' to open a PR
3. See error

**Expected behavior**
Opens a file / a browser with PR link

**Desktop (please complete the following information):**
 - OS: Debian WSL
 - commit: 8fd9dea6414a9b35af5b173854f90275229a460a

**Additional context**
i have a working fix at https://github.com/shinhs0506/lazygit/tree/wsl-support, would be happy to open a PR

### PR DESCRIPTION
**Pull request by shinhs0506**
fixes #1996

### CODE REVIEW

File: pkg/config/config_linux.go [hunk 1]
```diff
@@ -1,7 +1,27 @@
 package config
 
+import (
+	"io/ioutil"
+	"strings"
+)
+
+func isWSL() bool {
+	data, err := ioutil.ReadFile("/proc/sys/kernel/osrelease")
```
Comment (jesseduffield) [inline pkg/config/config_linux.go:9]: @shinhs0506 is this a large file? I wanna get an idea of the cost of reading it at startup

**Review by jesseduffield — CHANGES_REQUESTED** [review]
Just one question :) marking as changes requested for my own bookkeeping

File: pkg/config/config_linux.go (hunk 1 above)
Comment (shinhs0506) [reply pkg/config/config_linux.go:9]: The file is one line long

just as a reference, https://github.com/microsoft/WSL/issues/423#issuecomment-221627364

my `/proc/sys/kernel/osrelease` has the following content
`5.10.102.1-microsoft-standard-WSL2`

**Comment by jesseduffield** [comment]
@shinhs0506 looks like you've gotta run gofumpt on `pkg/config/config_linux.go` (see https://github.com/mvdan/gofumpt)

File: pkg/config/config_linux.go (hunk 1 above)
Comment (jesseduffield) [reply pkg/config/config_linux.go:9]: sounds good!

**Comment by shinhs0506** [comment]
should also fix #1513

**Comment by pirafrank** [comment]
Sorry to add to this, but I think there should be proper handling of setups where `xdg-open` is not installed, or a browser is not available in the terminal session lazygit runs in.

The error below appears on WSL2 using wezterm and (ctrl) clicking on links to try opening them in the default browser. lazygit crashes because of `xdg-open: command not found` (lazygit `0.40.2` on WSL2 Ubuntu 22.04 on Windows 11).

```txt
2024/09/03 12:33:31 An error occurred! Please create an issue at: https://github.com/jesseduffield/lazygit/issues

*errors.errorString bash: line 1: xdg-open: command not found

/home/runner/work/lazygit/lazygit/pkg/commands/oscommands/cmd_obj_runner.go:200 (0x9777db)
/home/runner/work/lazygit/lazygit/pkg/commands/oscommands/cmd_obj_runner.go:107 (0x97682c)
/home/runner/work/lazygit/lazygit/pkg/commands/oscommands/cmd_obj_runner.go:45 (0x975ea5)
/home/runner/work/lazygit/lazygit/pkg/commands/oscommands/cmd_obj.go:190 (0x97506f)
/home/runner/work/lazygit/lazygit/pkg/commands/oscommands/os.go:110 (0x979e77)
/home/runner/work/lazygit/lazygit/pkg/gui/information_panel.go:44 (0xaf41c6)
/home/runner/work/lazygit/lazygit/pkg/gui/keybindings.go:387 (0xaf810f)
/home/runner/work/lazygit/lazygit/pkg/gui/keybindings.go:373 (0xaf803b)
/home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:1424 (0x760fd9)
/home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:1392 (0x760e71)
/home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:1300 (0x760671)
/home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:798 (0x75ea0a)
/home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:752 (0x75e58b)
/home/runner/work/lazygit/lazygit/vendor/github.com/jesseduffield/gocui/gui.go:739 (0x75e2a5)
/home/runner/work/lazygit/lazygit/pkg/gui/gui.go:669 (0xaf0885)
/home/runner/work/lazygit/lazygit/pkg/gui/gui.go:675 (0xaf0dac)
/home/runner/work/lazygit/lazygit/pkg/utils/utils.go:108 (0x82b2c7)
/home/runner/work/lazygit/lazygit/pkg/gui/gui.go:674 (0xaf0cf4)
/home/runner/work/lazygit/lazygit/pkg/app/app.go:263 (0xb1afe9)
/home/runner/work/lazygit/lazygit/pkg/app/app.go:48 (0xb1af7e)
/home/runner/work/lazygit/lazygit/pkg/app/entry_point.go:150 (0xb1d286)
/home/runner/work/lazygit/lazygit/main.go:23 (0xb1eb1e)
/opt/hostedtoolcache/go/1.20.7/x64/src/runtime/internal/atomic/types.go:194 (0x437c87)
/opt/hostedtoolcache/go/1.20.7/x64/src/runtime/asm_amd64.s:1598 (0x467ec1)
```

A solution for this case may be to fallback to `x-www-browser` or do something similar to `gh browse` of GitHub CLI, which works in WSL2 and opens links in the default browser on Windows.

**Comment by LoliPain** [comment]
@pirafrank https://github.com/jesseduffield/lazygit/pull/1997#issuecomment-2326202786
> Also, `powershell.exe` may not be in `$PATH` because it has been disabled by the user

`sudo ln -s /mnt/c/Windows/System32/WindowsPowerShell/v1.0/powershell.exe /usr/local/bin/powershell.exe`

**Comment by LoliPain** [comment]
But generally, I agree that this PR is nonsense and should be reversed.

The true motivation of the author of this PR is apparently clear only to himself, in which case he can set himself a fork with this change instead of making these destructive changes in the master
Also, I found [THIS](https://github.com/jesseduffield/lazygit/issues/2757) which makes it even more hilarious reading about 

>  It is not recommended to use xdg-open as root. I feel like I would need to install this package as root to use with WSL Unbuntu. Isn't this dangerous?

Even tho `xdg-open` runs as common binary from the user you started it from, no matter how it was installed (I never would have thought that I would have to explain this to anyone)
```bash
❯ whoami
loli
❯ xdg-open . | ps -aux | grep xdg-open
loli     14921  0.0  0.0   2616  1748 pts/9    S+   06:18   0:00 /bin/sh /usr/bin/xdg-open .
```


To sum up: PR has done more harm than good, at least because most of everyone I know uses xdg-open along with WSL2, and now at least @pirafrank and me should have to use fix for this "fix"


At least those who vitally need to use open via `powershell start` can specify this in `~/.config/lazygit/config.yml`, so as to leave the most common scenario of working with xdg-open enabled by default

@jesseduffield Would you and you be so kind as to consider this matter?

**Comment by pirafrank** [comment]
I agree with the comment above. Also, I think this pkg may help https://github.com/cli/browser.

Customizing the `config.yml` is last resort imo, first because I think this should be handled in code, and second because a lot of people have their dotfiles repo shared cross-platform.
